### PR TITLE
[SOAR-4045]: Add connection validator

### PR DESCRIPTION
**Pull request by gusingh-r7**
## Description
- Add connection validator to fail validation when `cloud_ready:true` and `connection` has `credential_token`

### CODE REVIEW

**Review by jschipp-r7 — DISMISSED** [review]

**Review by jmcadams-r7 — DISMISSED** [review]

**Comment by mrinehart-r7** [comment]
@gusingh-r7 can you name the validator rule a bit more specific, maybe something like CloudReadyConnectionCredentialTokenValidator?

File: unit_test/test_validate_plugin.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 from icon_validator.rules.plugin_validators.version_validator import VersionValidator
 from icon_validator.rules.plugin_validators.version_pin_validator import VersionPinValidator
 from icon_validator.rules.plugin_validators.example_input_validator import ExampleInputValidator
+from icon_validator.rules.plugin_validators.connection_validator import ConnectionValidator
```
Comment (mrinehart-r7) [inline unit_test/test_validate_plugin.py:11]: Update import to match new name

File: unit_test/test_validate_plugin.py [hunk 2]
```diff
@@ -227,6 +228,12 @@ def test_example_input_validator_should_fail_when_not_all_exists(self):
         result = validate(directory_to_test, file_to_test, False, True, [ExampleInputValidator()])
         self.assertEqual(result, 1)
 
+    def test_cloud_ready_connection_credential_token_validator_should_fail(self):
+        directory_to_test = "plugin_examples/cloud_ready_connection_credential_token_validator"
+        file_to_test = "plugin.spec.yaml"
+        result = validate(directory_to_test, file_to_test, False, True, [ConnectionValidator()])
```
Comment (mrinehart-r7) [inline unit_test/test_validate_plugin.py:234]:
```suggestion
        result = validate(directory_to_test, file_to_test, False, True, [ConnectionValidator()])
```
Update to match new name

File: README.md [hunk 3]
```diff
@@ -62,6 +62,7 @@ be on your way to contributing!
 
 ## Changelog
 
+* 2.27.0 - Add Connection Validator
```
Comment (mrinehart-r7) [inline README.md:65]: Update to match new name (just be a bit more specific)

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

File: unit_test/test_validate_plugin.py (hunk 2 above)
Comment (gusingh-r7) [reply unit_test/test_validate_plugin.py]: Updated!

**Review by mrinehart-r7 — APPROVED** [review]

**Review by jmcadams-r7 — APPROVED** [review]

**Pull request merged**